### PR TITLE
Add watermarking

### LINKED ISSUE #4: add watermark
# Problem

We want to pull past data, and we don't want to pull data we already have. 

# Solution
- incorporate a watermark concept to only pull data, make requests for new past data we don't have 

# Blocking
this issue blocks #3 #2

### PR DESCRIPTION
**Pull request by littleforest**
closes #4 

## Description:
- Removes example foo model and table
- creates a watermark table to keep track of request urls and corresponding etags
- checks if request returns modified etag. If so, it will return updated data, if not it will return an empty hash

## QA:

- [x] In console:
```
[1] pry(main)> m = Meetup::Api.new(data_type: ["Women-Who-Code-Seattle"], options: {})
[2] pry(main)> m.get_response
```
should get a response.  

- [x] Run`m.get_response` again. Should get an empty hash.

- [x] Modify the `etag` of the watermark to something else: `Watermark.last.update(etag: "abc")`. Run `m.get_response` again. Should get results returned. Reload the last watermark, `etag` should be reset to actual value.

## Notes:

@WomenWhoCode/meet-maynard-reviewers

### CODE REVIEW

**Comment by isometimescode** [comment]
Looks like the review app failed because of:
```
NameError: uninitialized constant Foo
/app/db/seeds.rb:1:in `<top (required)>'
```

**Comment by littleforest** [comment]
I always forget about the seeds file! It would be nice if testing could pick that up somehow.  Anyway, updated!

File: lib/meetup/api.rb [hunk 1]
```diff
@@ -19,20 +19,23 @@ def initialize(data_type:, options:)
...
     # https://www.meetup.com/meetup_api/docs/#errors
     def get_response
-      url = build_url
-      MMLog.log.debug(url)
       throttle_wait
-      @response = HTTP.get(url)
-      set_throttle_values
-      get_body
+      @response = do_request
+      if @response.status.code == 304
```
Comment (isometimescode) [inline lib/meetup/api.rb]: Can we add a log message here? Just thinking it will make debugging weird responses easier in the future.

**Comment by isometimescode** [comment]
All tests worked!
👍 :shipit: ☀️

**Pull request merged**